tensorrt_llm/_torch/auto_deploy/custom_ops/attention/triton_paged_attention.py

```
@@ -1137,6 +1160,18 @@ def triton_paged_mha_with_cache(
     if num_prefill > 0:
         cu_seqlen = cu_seqlen_host[: num_prefill + 1].to(q.device, non_blocking=True)
         seq_len_with_cache = seq_len_with_cache_host[:num_prefill].to(q.device, non_blocking=True)
+        # For windowed cache_loc (VSWA), cap the cached-token portion of
+        # seq_len_with_cache to the actual pages available.  Without this,
+        # the context kernel computes page iteration bounds from global
+        # seq_len, overflowing the windowed cache_loc.
+        # seq_len_with_cache = cache_len + q_len, where cache_len is the
+        # number of prior-cached tokens.  Only cache_len needs capping.
+        q_lens = cu_seqlen[1 : num_prefill + 1] - cu_seqlen[:num_prefill]
+        page_counts = cu_num_pages[1 : num_prefill + 1] - cu_num_pages[:num_prefill]
+        max_cached = page_counts * kv_cache.shape[3]  # pages × page_size
```

Ruff (0.15.12)

[warning] Comment contains ambiguous `×` (MULTIPLICATION SIGN). Did you mean `x` (LATIN SMALL LETTER X)?

(RUF003)